.github/workflows/release.yml

```diff
@@ -0,0 +1,65 @@
+name: release
+
+on:
+  push:
+    tags:
+      - "v[0-9]+*"
+
+permissions:
+  contents: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    container:
+      image: ruby:2.6
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/vault-action@v2.4.2
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/apm-team/ci/apm-agent-ruby-rubygems-release apiKey | API_KEY ;
+      - name: RubyGems login
+        run: |
+          RUBY_HOME="${HOME}/.gem"
+          RUBY_CREDENTIALS_FILE="${RUBY_HOME}/credentials"
+          mkdir -p "${RUBY_HOME}"
+          echo '---' > "${RUBY_CREDENTIALS_FILE}"
+          echo ":rubygems_api_key: ${API_KEY}" >> ~/.gem/credentials
+          chmod 0600 "${RUBY_CREDENTIALS_FILE}"
+      - name: Install build system
+        run: gem install rake yard rspec
+      - run: rake release
+  update-branch:
+    needs:
+      - release
+    runs-on: ubuntu-latest
+    container:
+      image: ruby:2.6
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Git
+        uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - name: Install build system
+        run: gem install rake yard rspec
+      - run: rake release:update_branch
```

Comment:
What about using your forked repo and change the rake tast to target your fork?

https://github.com/elastic/apm-agent-ruby/blob/4e9f5300ac12deeb23dce185f5242d41f26afc5a/Rakefile#L26-L30 uses `main` and `4.x` branches, it should work as long as those branches exist in your fork

Reply:
I can think of two alternatives:
- `dry-run` when running the rake release in your fork
- change your fork so you can release to a new project using a different project name and version `0.0.999` (to avoid semver)

Reply:
For the latter, see https://github.com/elastic/apm-agent-ruby/pull/530 , it's the original request to support releases with Jenkins, for such I used a ruby gem hello world that helped me to test the CI changes in my fork from end to end